tools/create-release-candidate.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+# Copyright 2024 "Google LLC"
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e -o pipefail
+
+if ! type -P git-sed 1>/dev/null; then
+	echo "Must install git-extras package for git sed functionality"
+	exit 1
+fi
+
+if ! type -P gh 1>/dev/null; then
+	echo "Must install GitHub CLI tool for command line API access"
+	exit 1
+fi
+
+GITDIR=$(mktemp -d)
+trap 'rm -rf ${GITDIR}' EXIT
+
+if [ "$#" -le 1 ]; then
+	BUMP_MAJOR=0
+	BUMP_MINOR="${1:-1}"
+	BUMP_MAJOR=0
+elif [ "$#" -eq 3 ]; then
+	BUMP_MAJOR="$1"
+	BUMP_MINOR="$2"
+	BUMP_MAJOR="$3"
+else
+	echo "Must provide 1 argument to bump minor version or all 3 for any other version change"
+	exit 1
+fi
+
+OLD_VERSION=$(gh release list -R GoogleCloudPlatform/hpc-toolkit -L 1 --json tagName | jq -r '.[] | .tagName')
+OLD_MAJOR=$(echo "${OLD_VERSION}" | cut -f1 -d. | sed 's,v,,')
+OLD_MINOR=$(echo "${OLD_VERSION}" | cut -f2 -d.)
+OLD_PATCH=$(echo "${OLD_VERSION}" | cut -f3 -d.)
+
+NEW_MAJOR=$((OLD_MAJOR + BUMP_MAJOR))
+NEW_MINOR=$((OLD_MINOR + BUMP_MINOR))
+NEW_PATCH=$((OLD_PATCH + BUMP_PATCH))
+NEW_VERSION="v${NEW_MAJOR}.${NEW_MINOR}.${NEW_PATCH}"
+
+RC_BRANCH=release-candidate
+V_BRANCH="version/${NEW_VERSION}"
+
+gh repo clone GoogleCloudPlatform/hpc-toolkit "${GITDIR}" -- --single-branch --branch develop --depth 1
+cd "${GITDIR}"
+git switch -c "${RC_BRANCH}" develop
+git push -u origin "${RC_BRANCH}"
+git switch -c "${V_BRANCH}" "${RC_BRANCH}"
+git sed "v${OLD_MAJOR}\.${OLD_MINOR}\.${OLD_PATCH}" "v${NEW_VERSION}" -- **/*.go **/versions.tf
+git add -u
+git commit -m "Increase version to ${NEW_VERSION}"
+git push -u origin "${V_BRANCH}"
+gh pr create --base "${RC_BRANCH}" --head "${V_BRANCH}" \
+	--draft \
+	--title "Increase version to ${NEW_VERSION}" \
+	--body "Set release-candidate to version ${NEW_VERSION}"
```

Comment:
Last line could be an invocation of `babysit --pr ${pr_num} --auto -c 1`

Reply:
My main concern here would be setting an expectation to the user that the script will run in background for a long time (e.g. you should run this under tmux on a remote VM). I might have script suggest the command or execute it optionally.